mantidproject/mantid
# Allow Selection of Flood and Polarization Efficiencies Workspace by File Path

**Description of work.**
Added text entry boxes to allow specification of a filepath for the polarisation efficiencies workspace and the flood corrections workspace. This will speed up the workflow by removing the need to load the file separately, and allows the information to be held in the batch. Making it a more complete saving of the GUI state. 

**To test:**
1. Talk to Conor or Rachel about getting the corrections files. It's not public, so it can't be put here. 
3. Boot up the Refl GUI
4. Switch the instrument to POLREF.
5. Go to the Experiment Settings Tab
6. Enable polarisation corrections 
7. Check the combo box gets enabled on workspace and becomes a text edit on filepath. 
9. Enter the file path to the downloaded workspace
11. Go back to the Runs tab
12. Enter `31835` into the Run column (not the group, the second box down) and `1.0` into the angle column
13. Click process
14. Check that the `IvsLam_31835` workspace group contains a workspace ending with `_--` and `_++`
15. Do similar with the flood workspaces.

Fixes #34720 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
